src/Lykke.Service.SyntheticFiatFeed.AzureRepositories/SimBaseInstrumentSettingRepository.cs

```diff
@@ -0,0 +1,129 @@
+
+        public SimBaseInstrumentSettingRepository(INoSQLTableStorage<SimBaseInstrumentSettingEntity> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        }
+
+        public async Task<IReadOnlyList<ISimBaseInstrumentSetting>> GetAllSettings()
+        {
+            if (_cache == null)
+            {
+                var data = await _tableStorage.GetDataAsync();
+                _cache = data.ToDictionary(e => e.BaseAssetPair, e => e);
+            }
+
+            return _cache.Values.ToList();
+        }
+
+        public async Task AddOrUpdateSettings(ISimBaseInstrumentSetting setting)
+        {
+            if (_cache.TryGetValue(setting.BaseAssetPair, out var entity))
+            {
+                entity.CopyFrom(setting);
+                await _tableStorage.ReplaceAsync(entity);
+            }
+            else
+            {
+                entity = new SimBaseInstrumentSettingEntity(setting);
+                await _tableStorage.InsertOrReplaceAsync(entity);
+                _cache[entity.BaseAssetPair] = entity;
+            }
+        }
+    }
+
+    public class SimBaseInstrumentSettingEntity : AzureTableEntity, ISimBaseInstrumentSetting
+    {
+        private IReadOnlyList<string> _sourceExchange = new List<string>();
+        private IReadOnlyList<ILinkedInstrumentSettings> _crossInstrument = new ILinkedInstrumentSettings[]{};
+
+        public SimBaseInstrumentSettingEntity()
+        {
+        }
+
+        public SimBaseInstrumentSettingEntity(ISimBaseInstrumentSetting setting)
+        {
+            PartitionKey = GeneratePartitionKey(setting.BaseAssetPair);
+            RowKey = GenerateRowKey(setting.BaseAssetPair);
+            BaseAssetPair = setting.BaseAssetPair;
+            _sourceExchange = setting.SourceExchange;
+            CountPerSecond = setting.CountPerSecond;
+            PriceAccuracy = setting.PriceAccuracy;
+            FakeVolume = setting.FakeVolume;
+            _crossInstrument = setting.CrossInstrument;
+            DangerChangePriceKoef = setting.DangerChangePriceKoef;
+            Order = setting.Order;
+        }
+
+        public void CopyFrom(ISimBaseInstrumentSetting setting)
+        {
+            _sourceExchange = setting.SourceExchange;
+            CountPerSecond = setting.CountPerSecond;
+            PriceAccuracy = setting.PriceAccuracy;
+            FakeVolume = setting.FakeVolume;
+            _crossInstrument = setting.CrossInstrument;
+            DangerChangePriceKoef = setting.DangerChangePriceKoef;
+            Order = setting.Order;
+        }
+
+        public static string GeneratePartitionKey(string baseAssetPair)
+        {
+            return baseAssetPair;
+        }
+
+        public static string GenerateRowKey(string baseAssetPair)
+        {
+            return baseAssetPair;
+        }
+
+        public string BaseAssetPair { get; set; }
+
+        public string SourceExchange
```

Comment:
I suppose you can use "JsonValueSerializer" attribute instead of separate property